controllers/policies/policyserver_controller.go

```diff
@@ -152,71 +192,43 @@ func (r *PolicyServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	err = ctrl.NewControllerManagedBy(mgr).
 		For(&policiesv1alpha2.PolicyServer{}).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Watches(&source.Kind{Type: &policiesv1alpha2.ClusterAdmissionPolicy{}}, handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
-			// The watch will trigger twice per object change; once with the old
-			// object, and once the new object. We need to be mindful when doing
-			// Updates since they will invalidate the newever versions of the
-			// object.
-			policy, ok := object.(*policiesv1alpha2.ClusterAdmissionPolicy)
-			if !ok {
-				r.Log.Error(err, "object is not type of ClusterAdmissionPolicy: %#v", policy)
-				return []ctrl.Request{}
-			}
-
-			if policy.Status.PolicyStatus == "" {
-				policy.Status.PolicyStatus = policiesv1alpha2.ClusterAdmissionPolicyStatusUnscheduled
-				err := r.Reconciler.UpdateAdmissionPolicyStatus(context.Background(), policy)
-				if err != nil {
-					r.Log.Error(err, "cannot update status of policy "+policy.Name)
-					return []ctrl.Request{}
+		Watches(&source.Kind{Type: &policiesv1alpha2.ClusterAdmissionPolicy{}}, handler.Funcs{
+			CreateFunc: func(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+				q.Add(watchClusterAdmissionPolicyj(r, e.Object))
+			},
+			UpdateFunc: func(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
```

Comment:
```suggestion
			UpdateFunc: func(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
```

Reply:
in other places you use just `q` I don't really have a preference for `q` or `queue` but it should be the same in all funcs